### PR TITLE
fix: honor cursor blink rate

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/10668.

Ensures that WebPreferences honor the system cursor blink rate on creation.

cc @zcbenz @nornagon @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where cursor blink rate was not honored.

### CODE REVIEW

File: shell/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -458,6 +468,23 @@ void WebContents::InitWithSessionAndOptions(
...
+  views::LinuxUI* linux_ui = views::LinuxUI::instance();
+  if (linux_ui)
+    prefs->caret_blink_interval = linux_ui->GetCursorBlinkInterval();
+#endif
+
+#if defined(OS_WIN)
+  static const size_t system_value = ::GetCaretBlinkTime();
+  if (system_value != 0 && system_value != INFINITE)
+    prefs->caret_blink_interval =
+        base::TimeDelta::FromMilliseconds(system_value);
```
Comment (ckerr) [inline shell/browser/api/atom_api_web_contents.cc]: _(minor)_ One thing that might be nice to copy from [TextField::GetCaretBlinkInterval](https://cs.chromium.org/chromium/src/ui/views/controls/textfield/textfield.cc?sq=package:chromium&dr=CSs&g=0&l=266) is that if `::GetCaretBlinkTime` fails, it uses a 500msec blink as a fallback

e.g.

```cpp
const auto system_msec = ::GetCaretBlinkTime();
if (system_msec == INFINITE)
  prefs->caret_blink_interval = base::TimeDelta()
else if (system_msec == 0)
  prefs->caret_blink_interval = base::TimeDelta::FromMilliseconds(500);
else
  prefs->caret_blink_interval = base::TimeDelta::FromMilliseconds(system_msec);
```

File: shell/browser/api/atom_api_web_contents.cc [hunk 2]
```diff
@@ -458,6 +468,23 @@ void WebContents::InitWithSessionAndOptions(
   prefs->subpixel_rendering = params->subpixel_rendering;
 #endif
 
+#if defined(OS_MACOSX)
+  base::TimeDelta interval;
+  if (ui::TextInsertionCaretBlinkPeriod(&interval))
+    prefs->caret_blink_interval = interval;
```
Comment (ckerr) [inline shell/browser/api/atom_api_web_contents.cc]: 500msec would be a good fallback default here too

Comment (codebytere) [reply shell/browser/api/atom_api_web_contents.cc]: macOS automatically falls back to 500 :)

File: shell/browser/api/atom_api_web_contents.cc [hunk 3]
```diff
@@ -100,6 +100,16 @@
 
 #if !defined(OS_MACOSX)
 #include "ui/aura/window.h"
+#else
+#include "ui/base/cocoa/defaults_utils.h"
+#endif
+
+#if defined(TOOLKIT_VIEWS)
+#include "ui/views/controls/textfield/textfield.h"
```
Comment (miniak) [inline shell/browser/api/atom_api_web_contents.cc]: we don't need this unless we use `views::Textfield::GetCaretBlinkInterval();`

**Comment by miniak** [comment]
@codebytere can we use `views::Textfield::GetCaretBlinkInterval();`? it wraps `GetCaretBlinkTime`

**Comment by codebytere** [comment]
@miniak it does wrap that method, but it also wraps things for mac and so for clarity & maintainability i’d prefer we explicitly set all three ourselves since we’d need to call the linux one separately regardless.

Perhaps we could consider upstreaming the linux one into that call and then using that in the future?

**Pull request merged**